MovieApp/PresentationLayer/HomeScreen/CategorisedCollection/CategorisedCollectionView.swift

```diff
@@ -6,12 +6,24 @@ class CategorisedCollectionView: UIView {
     let defaultSpacing: CGFloat = 10
 
     var onCategoryChanged: (Int) -> Void = { _ in }
+    var onMovieFavorited: (Int) -> Void = { _ in }
 
     var titleLabel: UILabel!
-    var optionsView: ButtonBarView!
+    var categoriesView: ButtonBarView!
     var movieCollectionView: UICollectionView!
+    var currentlySelectedCategory: OptionViewModel {
```

Comment:
return optional

Reply:
```
extension Array {

    /**
     Allows for returning of 'nil' values in case a specified index is out of bounds.

     - parameter index: Index of array element to return.
     - returns: Array element or 'nil' if index is out of array bounds.
     */
    func at(_ index: Int) -> Element? {
        guard index >= 0 && index < count else { return nil }

        return self[index]
    }
}
```